vizro-ai/docs/pages/user-guides/vizro-ai-langchain-guide.md

```diff
@@ -0,0 +1,193 @@
+# Using Vizro-AI methods as LangChain tools
+
+This guide demonstrates how to integrate Vizro-AI's chart and dashboard generation capabilities as LangChain tools. This integration enables you to use Vizro-AI's functionality within a larger LangChain application.
+
+## 1. Set up the environment
+
+First, import the required libraries and prepare your data:
+
+```python
+from copy import deepcopy
+from typing import Annotated, Any
+
+import pandas as pd
+import vizro.plotly.express as px
+from langchain_core.runnables import chain
+from langchain_core.tools import InjectedToolArg, tool
+from langchain_openai import ChatOpenAI
+from vizro_ai import VizroAI
+
+# Load sample data
+df = px.data.gapminder()
+dfs = [df]
+
+# Initialize the LLM
+llm = ChatOpenAI(model="gpt-4o-mini")
+```
+
+## 2. Define LangChain tools
+
+Basic tools only take string as input and output. Vizro-AI takes Pandas dataframes as input and it's not cost-efficient and secure to pass the full data to a LLM. In this case, [bind the Pandas dataframes at run time](https://python.langchain.com/v0.2/docs/how_to/tool_runtime/) is more suitable.
```

Comment:
```suggestion
Vizro-AI takes Pandas dataframes as input, but it's not cost-efficient and secure to pass the full data to a LLM since basic tools only take strings as input and output. In this step, [bind the Pandas dataframes at run time](https://python.langchain.com/v0.2/docs/how_to/tool_runtime/).
```

Reply:
Current steps 2 and 3 together make the approach complete. Now I reword the sentence as "The recommended approach is to exclude DataFrame parameters from the tool's schema (which is the step2) and instead bind them at runtime (which is the step3) using LangChain's runtime binding feature". Would this be more clear?